### ISSUE TITLE
table_name

### ISSUE THREAD

**Issue by rekoy88** [Infinidat/infi.clickhouse_orm#88]
Hello guys, tell me how to set your own table name without changing the class name

**Comment by ishirav** [comment]
See the documentation: https://github.com/Infinidat/infi.clickhouse_orm/blob/develop/docs/models_and_databases.md#table-names

**Comment by rekoy88** [comment]
Yep, but class which is inherited from Models does not allow to add own attributes to __init__

**Comment by ishirav** [comment]
I'm not sure I understand what you mean. You asked about using a different table name for a model (instead of the default name), this is how you do it:
```
class MyModel(Model):

    date = DateField()
    f1 = Int32Field()
    f2 = StringField()

    engine = MergeTree('date', ('date',))

    @classmethod
    def table_name(cls):
        return 'my_table_name'
```

**Comment by ishirav** [comment]
Now I understand what you're trying to do. You have several identical tables, and you want to choose which one to insert into? You could create a base Model that contains the table structure (fields), and then create a subclass for each table name. For example:
```
class Person(Model):
    name = StringField()
    birthday = DateField()

class AmericanPerson(Person):
    pass

class CanadianPerson(Person):
    pass

class MexicanPerson(Person):
    pass
```

This way you'll have several identical tables - americanperson, canadianperson, mexicanperson.
And now each person that you want to insert into the database, you insert via the appropriate Model class.
